### PR TITLE
Added icon on the left and click-event to tag

### PR DESCRIPTION
**Pull request by Timoms**
## Reason

Tags are a beautiful way to show a specific type. With an icon on the left you can create different categories like "car" and "motorcycle" using different icons. That way the user can exactly see what type of category this tag applies to.

Click events on the other hand are useful when implementing e.g. taglists where the user can interact with different tags to emit specific tasks (like apply a search filter based on the tag clicked).

## Proposed Changes

- Added icon on the left using:

`<tag icon="cloud-tags" closable attached>Tag with icon on the left!</tag>`

- Added click event to tag content using

`<tag icon="cloud-tags" @click="testClick">Tag with icon on the left!</tag>`

### CODE REVIEW

File: src/components/tag/Tag.vue [hunk 1]
```diff
@@ -33,7 +34,8 @@
         v-else
         class="tag"
         :class="[type, size, { 'is-rounded': rounded }]">
-        <span :class="{ 'has-ellipsis': ellipsis }">
+        <b-icon custom-class="" v-if="icon" :icon="icon" :size="size" :type="iconType" :pack="iconPack" />
```
Comment (jtommy) [inline src/components/tag/Tag.vue:37]: Please remove `custom-class`

**Review by jtommy — COMMENTED** [review]
Please update API docs, tag.js file and regen snapshot running tests

File: src/components/tag/Tag.vue (hunk 1 above)
Comment (Timoms) [reply src/components/tag/Tag.vue:37]: Thanks for letting me know, going to do the changes in the upcoming days.

**Comment by mouadTaoussi** [comment]
The snapshot is no longer comparable to the newest version of the component, I think you should rewrite it, it occurs lot of failing tests.

**Pull request merged**